experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/client/Client.scala

```diff
@@ -162,44 +168,118 @@ class Client(configuration:AppConfig, yarnConf: YarnConfiguration, yarnClient: Y
     })
   }
 
-  def monitorAM(appContext: ApplicationSubmissionContext): Unit = {
-    val appId = appContext.getApplicationId
+  def delay(appId: ApplicationId): (ApplicationReport, YarnApplicationState) = {
+    Thread.sleep(1000)
+    val appReport = yarnClient.getApplicationReport(appId)
+    val appState = appReport.getYarnApplicationState
+    (appReport, appState)
+  }
+
+  def monitor(appId: ApplicationId): Unit = {
     var appReport = yarnClient.getApplicationReport(appId)
     var appState = appReport.getYarnApplicationState
-    while (appState != YarnApplicationState.FINISHED &&
-      appState != YarnApplicationState.KILLED &&
-      appState != YarnApplicationState.FAILED) {
-      Thread.sleep(1000)
-      appReport = yarnClient.getApplicationReport(appId)
-      appState = appReport.getYarnApplicationState
+    val monitor = getEnv("monitor").toBoolean
+    var watch = true
+
+    while(watch) {
+      appState match {
+        case YarnApplicationState.FINISHED =>
+          LOG.info(s"Application $appId finished with state $appState at ${appReport.getFinishTime}")
+          watch = false
+        case YarnApplicationState.KILLED =>
+          LOG.info(s"Application $appId finished with state $appState at ${appReport.getFinishTime}")
+          watch = false
+        case YarnApplicationState.FAILED =>
+          LOG.info(s"Application $appId finished with state $appState at ${appReport.getFinishTime}")
+          watch = false
+        case YarnApplicationState.SUBMITTED =>
+          val (ar, as) = delay(appId)
+          appReport = ar
+          appState = as
+        case YarnApplicationState.ACCEPTED =>
+          val (ar, as) = delay(appId)
+          appReport = ar
+          appState = as
+        case YarnApplicationState.RUNNING =>
+          monitor match {
```

Comment:
See above note about whether we watch the app until a terminate state is reached or just until the app is RUNNING.